.github/workflows/pre-commit.yml

```diff
@@ -14,3 +14,6 @@ jobs:
         uses: actions/setup-python@v5
       - name: Run pre-commit
         uses: pre-commit/action@v3.0.1
+      - name: Use the Command Line Tools
+        run: |
+          sudo xcode-select -s /Library/Developer/CommandLineTools
```

Comment:
I might be wrong, but I'm not sure this will work because within that dir there are a bunch of symlinks:
```
$ l /Library/Developer/CommandLineTools/SDKs/      
total 0
drwxr-xr-x  7 root  wheel   224B May 20 13:36 .
drwxr-xr-x  5 root  wheel   160B May 20 13:36 ..
lrwxr-xr-x  1 root  wheel    14B May 20 13:36 MacOSX.sdk -> MacOSX14.4.sdk
drwxr-xr-x  7 root  wheel   224B May 12  2023 MacOSX13.3.sdk
lrwxr-xr-x  1 root  wheel    14B May 20 13:35 MacOSX13.sdk -> MacOSX13.3.sdk
drwxr-xr-x  7 root  wheel   224B Feb 19 18:28 MacOSX14.4.sdk
lrwxr-xr-x  1 root  wheel    14B May 20 13:35 MacOSX14.sdk -> MacOSX14.4.sdk
```

I think this is just selecting the currently enabled SDK.

Reply:
I think this just prevents the use of `/Applications/Xcode.app/Contents/Developer` where the C++ SDK might be missing.

We might still end up using different SDKs in different stages, but at least not an empty one.

Reply:
Ok, this is simpler than my fix so if it works we can try this first.